adot-testbed/app/src/test/java/software/amazon/adot/testbed/LogsTests.java

```diff
@@ -0,0 +1,150 @@
+package software.amazon.adot.testbed;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import com.github.rholder.retry.RetryerBuilder;
+import com.github.rholder.retry.StopStrategies;
+import com.github.rholder.retry.WaitStrategies;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.MountableFile;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.GetLogEventsRequest;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Testcontainers(disabledWithoutDocker = true)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class LogsTests {
+    private static final String TEST_IMAGE = System.getenv("TEST_IMAGE") != null && !System.getenv("TEST_IMAGE").isEmpty()
+        ? System.getenv("TEST_IMAGE")
+        : "public.ecr.aws/aws-otel-test/adot-collector-integration-test:latest";
+    private final Logger collectorLogger = LoggerFactory.getLogger("collector");
+    private static final String uniqueID = UUID.randomUUID().toString();
+
+    private GenericContainer<?> collector;
+
+    LogsTests() throws Exception {
+    }
+
+    private GenericContainer<?> createAndStartCollector(String configFilePath, String logFilePath, String uniqueLogStreamName) throws IOException {
+
+        // Create an environment variable map
+        Map<String, String> envVariables = new HashMap<>();
+        envVariables.put("LOG_STREAM_NAME", uniqueLogStreamName);
+        //Set credentials
+        envVariables.put("AWS_REGION", System.getenv("AWS_REGION"));
+        envVariables.put("AWS_ACCESS_KEY_ID", System.getenv("AWS_ACCESS_KEY_ID"));
+        envVariables.put("AWS_SECRET_ACCESS_KEY", System.getenv("AWS_SECRET_ACCESS_KEY"));
+        envVariables.put("AWS_SESSION_TOKEN", System.getenv("AWS_SESSION_TOKEN"));
```

Comment:
session token is optional. so you should only inject this variable if it is not null.

Reply:
Agreed

Reply:
you forgot to remove line [64](https://github.com/aws-observability/aws-otel-test-framework/pull/1426/files#diff-b6b213810cf9bd4029972a83c70ccf32f71e4f4eb1e32cc5b6b90dd5bcf7961aR64)  since you are optionally injecting the `AWS_SESSION_TOKEN` on line 66.